nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseCEF.java

```diff
@@ -0,0 +1,323 @@
+import java.net.InetAddress;
+import java.text.SimpleDateFormat;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+
+@EventDriven
+@SideEffectFree
+@SupportsBatching
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@Tags({"logs", "cef", "attributes", "system", "event", "message"})
+@CapabilityDescription("Parses the contents of a CEF formatted message and adds attributes to the FlowFile for " +
+        "headers and extensions of the parts of the CEF message.\n" +
+        "Note: This Processor expects CEF messages WITHOUT the syslog headers (i.e. starting at \"CEF:0\"")
+@WritesAttributes({@WritesAttribute(attribute = "cef.header.version", description = "The version of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceVendor", description = "The Device Vendor of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceProduct", description = "The Device Product of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceVersion", description = "The Device Version of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.deviceEventClassId", description = "The Device Event Class ID of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.name", description = "The name of the CEF message."),
+    @WritesAttribute(attribute = "cef.header.severity", description = "The severity of the CEF message."),
+    @WritesAttribute(attribute = "cef.extension.*", description = "The key and value generated by the parsing of the message.")})
+@SeeAlso({ParseSyslog.class})
+
+public class ParseCEF extends AbstractProcessor {
+
+    // There should be no date format other than internationally agreed formats...
+    // flowfile-attributes uses Java 8 time to parse data (as Date  objects are not timezoned)
+    private final static DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+    // for some reason Jackson doesnt seem to be able to use DateTieFormater
+    // so we use a SimpleDateFormat to format within flowfile-content
+    private final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+
+    // add a TZ object to be used by flowfile-attribute routine
+    private String tzId = null;
+
+    // Add serializer and mapper
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    public static final String DESTINATION_CONTENT = "flowfile-content";
+    public static final String DESTINATION_ATTRIBUTES = "flowfile-attribute";
+    public static final PropertyDescriptor FIELDS_DESTINATION = new PropertyDescriptor.Builder()
+        .name("FIELDS_DESTINATION")
+        .displayName("Parsed fields destination")
```

Comment:
happy to change that. I have been using all caps matching the property descriptor name within code but happy to adjust in case a naming convention already exists.